src/game/AI/BaseAI/PetAI.cpp

```diff
@@ -40,9 +40,8 @@ PetAI::PetAI(Creature* creature) : UnitAI(creature), m_creature(creature), inCom
     m_AllySet.clear();
     UpdateAllies();
 
-    /* hunter pets attack from behind (tbc and later)
-    if (((Pet*)creature)->getPetType() == HUNTER_PET)
-        m_attackAngle = M_PI_F; */
+    if (((Pet*)creature)->getPetType() == HUNTER_PET && sWorld.getConfig(CONFIG_BOOL_PET_ATTACK_FROM_BEHIND))
```

Comment:
@Warlockbugs - this should be good now, yea?

Reply:
Any pet spawned by a player should do this, not just hunter pets.

Reply:
Yeah, that is what I was thinking as well, but guardians behave like their own entities, you don't control them, I think it's only for controlled pets. So either remove the (Pet*)creature)->getPetType() == HUNTER_PET part entirely or replace with (Pet*)creature)->isControlled()?

Reply:
`static_cast<Pet*>(creature)->isControlled()` :smiley:

Reply:
Hmm, why dynamic and not static cast?

Reply:
In this case your safe downcasting is rather worthless, because you're using the result unconditionally. If ```creature``` isn't really a pointer to Pet, you'll get a nullptr which you then dereference. You could as well use a static_cast then, because the end result is going to be the same anyway: :boom: